### PR TITLE
Create Separate Scan Now Modal

### LINKED ISSUE #1105: Create Separate Scan Now Modal
Create a new modal that can run a stratagem scan if all preconditions are met. Ensure it is separate from the timer configuration.

### PR DESCRIPTION
**Pull request by johnsonw**
Fixes #1105.

Create a new modal that can run a stratagem scan if all preconditions are met. Ensure it is separate from the timer configuration.

Signed-off-by: Will Johnson <wjohnson@whamcloud.com>

### CODE REVIEW

**Comment by johnsonw** [comment]
image

image

**Comment by johnsonw** [comment]
Since we are moving this out should we close this PR?

**Comment by jgrund** [comment]
No. I'm going to use it as basis for refresh.

**Comment by jgrund** [comment]
Added modal for immediate scanning:

Screen Shot 2019-08-12 at 9 55 50 PM

**Comment by jgrund** [comment]
Also included a number of refactors. Some of the larger ones include updating the duration picker to internally store the value as a `Option<u64>` instead of `String`, and adding the conversion method to ms directly to the duration_picker model.

**Comment by jgrund** [comment]
scan-now

File: iml-wasm-components/iml-stratagem/src/lib.rs [hunk 1]
```diff
@@ -97,68 +97,21 @@ fn max_value_validation(ms: u64, unit: iml_duration_picker::Unit) -> Option<Stri
...
-        };
-
         Some(StratagemUpdate {
-            id,
+            id: self.id?,
             filesystem: self.fs_id,
-            interval,
-            report_duration,
-            purge_duration,
+            interval: self.run_config.value_as_ms()?,
```
Comment (johnsonw) [inline iml-wasm-components/iml-stratagem/src/lib.rs]: nice :)

File: iml-wasm-components/iml-stratagem/src/lib.rs [hunk 2]
```diff
@@ -248,41 +173,31 @@ pub fn update(msg: Msg, model: &mut Model, orders: &mut impl Orders<Msg>) {
 
             model.validate();
 
-            if model.config_valid() {
-                model.enable_stratagem_button = model.create_enable_stratagem_model();
+            model.enable_stratagem_button = if model.config_valid() {
```
Comment (johnsonw) [inline iml-wasm-components/iml-stratagem/src/lib.rs]: What do you think about having `model.create_enable_stratagem_model()` return an option?

**Comment by johnsonw** [comment]
I can't +1 since I opened the review but I don't see anything blocking. LGTM +1

File: iml-wasm-components/iml-duration-picker/duration-picker.rs [hunk 3]
```diff
@@ -49,37 +60,30 @@ pub enum Msg {
...
         }
         Msg::WatchChange => model.watching.update(),
         Msg::InputChange(ev) => {
             let target = ev.target().unwrap();
             let input_el = seed::to_input(&target);
 
-            model.value = input_el.value().trim().to_string();
+            let value = input_el.value_as_number();
+
+            model.value = if value.is_nan() {
```
[the remark this answers is not in the document]
Comment (jgrund) [reply iml-wasm-components/iml-duration-picker/duration-picker.rs]: If the value is empty it's going to be `NaN`. I think this is as close as we can get.

File: iml-wasm-components/iml-stratagem/src/enable_stratagem_button.rs [hunk 4]
```diff
@@ -59,7 +57,7 @@ fn enable_stratagem(model: &Model) -> impl Future<Item = Msg, Error = Msg> {
 pub fn view(model: &Option<Model>) -> Node<Msg> {
     let btn = bs_button::btn(
         class![bs_button::BTN_PRIMARY],
-        vec![Node::new_text("Enable")],
+        vec![Node::new_text("Enable Interval")],
```
[the remark this answers is not in the document]
Comment (jgrund) [reply iml-wasm-components/iml-stratagem/src/enable_stratagem_button.rs]: I'll update.

**Pull request merged**